manifests/sar_metric.pp

```diff
@@ -0,0 +1,59 @@
+define puppet_metrics_collector::sar_metric (
+  String                    $output_dir,
+  String                    $scripts_dir,
+  Enum['absent', 'present'] $metric_ensure  = 'present',
+  String                    $metrics_type   = $title,
+  String                    $cron_minute    = '*/5',
+  Integer                   $retention_days = 90,
+  Integer                   $polling_frequency_seconds = 1,
+  Integer                   $collection_frequency = 5, #minutes
+  String                    $metric_script_file = 'generate_system_metrics',
+) {
+
+  $metrics_output_dir = "${output_dir}/${metrics_type}"
+
+  $_metric_ensure = $metric_ensure ? {
+    'present' => directory,
+    'absent'  => absent,
+  }
+
+  file { $metrics_output_dir :
+    ensure => $_metric_ensure,
+  }
+
+  $script_file_name = "${scripts_dir}/${metric_script_file}"
+  $file_interval_seconds = $collection_frequency * 60
```

Comment:
collection_interval_seconds  ?? it is unclear why the name change to `file`.

Reply:
because at this level I could change the name without affecting the ability to merge system into init.pp